### PR TITLE
auto case OCP-69092

### PR DESCRIPTION
**Pull request by jinyunma**

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

**Comment by jinyunma** [comment]
cc @gpei @jianlinliu to review

File: ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        echo "ERROR: Not found shared tags ${expected_shared_tags} on vnet ${vnet_id}"
+        no_critical_check_result=1
+    fi
+
+    if [[ ! -z "${AZURE_VNET_TAGS}" ]]; then
+        echo "check custom tags ${AZURE_VNET_TAGS} are not overriden"
+        for tag in ${AZURE_VNET_TAGS}; do
+            tag_content=$(echo $tag | awk -F'=' '{printf "\"%s\": \"%s\"",$1, $2}')
+            echo "expected custom tag: ${tag_content}"
+            if grep -Fq "${tag_content}" ${vnet_list_file}; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh:85]: Even in the older version of OCP, once the custom tag is added, the tag will not be touched, too. right?

File: ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    exit 1
+fi
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+no_critical_check_result=0
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh:58]: How about rename it to `critical_check_failure_count`?

File: ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+#nsg rule 'apiserver_in' check
+ocp_minor_version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f2)
+if (( ${ocp_minor_version} < 15 )); then
+    echo "Shared Tags on vnet checking is only applicable on 4.15+, skip the check!"
+    exit 0
+else
+    echo "Checking that shared tags are added on existing vnet"
+    vnet_list_file=$(mktemp)
+    expected_shared_tags="\"kubernetes.io_cluster.${INFRA_ID}\": \"shared\""
```
[the remark this answers is not in the document]
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh:67]: could not, this is new added on 4.15 through https://github.com/openshift/installer/pull/7611.

File: ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh (hunk 1 above)
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh:85]: It is only supported on 4.15 to add ocp shared tags on existing vnet.  
Here make sure that customer tags could not be overridden when ocp shared tag is added.

File: ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh (hunk 2 above)
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-commands.sh]: updated

File: ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-ref.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    default: ""
+    documentation: |-
+      Format: "key1=value1 key2=value2"
+      Add customer tags on existing vnet to test shared tags added by installer is not overriden.
+  - name: EXIT_ON_INSTALLER_CHECK_FAIL
+    default: "no"
+    documentation: to determine if exit with error when non critical check failed.
+  dependencies:
+  - name: "release:latest"
+    env: RELEASE_IMAGE_LATEST_FROM_BUILD_FARM
```
[the remark this answers is not in the document]
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-ref.yaml]: removed

File: ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-ref.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  env:
+  - name: AZURE_VNET_TAGS
+    default: ""
+    documentation: |-
+      Format: "key1=value1 key2=value2"
+      Add customer tags on existing vnet to test shared tags added by installer is not overriden.
+  - name: EXIT_ON_INSTALLER_CHECK_FAIL
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-ref.yaml:18]: If this ENV is not required, so we can remove it, too.

Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/azure/vnet/tag/cucushift-installer-check-azure-vnet-tag-ref.yaml]: yes, removed.

**Comment by jinyunma** [comment]
/pj-rehearse ack

cc @liangxia  to review

**Pull request merged**